src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -449,6 +469,11 @@ def _unsupported_type_error(dimensions):
                 else:
                     _unsupported_type_error(dimensions)
 
+            elif isinstance(dim, Fortran2003.Assumed_Size_Spec):
+                raise NotImplementedError(
+                    "Could not process {0}. Assumed-size arrays"
+                    " are not supported.".format(dimensions))
+
             else:
                 raise InternalError(
                     "Reached end of loop body and {0} has"
```

Comment:
Not yours probably but please could you s/and {0}/and array-shape specification {0}/

Reply:
(probably mine from a past PR :)) Added the suggested text